osu.Game/Storyboards/Commands/StoryboardCommandGroup.cs

```diff
@@ -0,0 +1,130 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System.Collections.Generic;
+using System.Linq;
+using Newtonsoft.Json;
+using osu.Framework.Graphics;
+using osu.Framework.Lists;
+using osuTK;
+using osuTK.Graphics;
+
+namespace osu.Game.Storyboards.Commands
+{
+    public class StoryboardCommandGroup
+    {
+        private readonly SortedList<StoryboardCommand<float>> x = new SortedList<StoryboardCommand<float>>();
+
+        public IReadOnlyList<StoryboardCommand<float>> X => x;
+
+        private readonly SortedList<StoryboardCommand<float>> y = new SortedList<StoryboardCommand<float>>();
+
+        public IReadOnlyList<StoryboardCommand<float>> Y => y;
+
+        private readonly SortedList<StoryboardCommand<float>> scale = new SortedList<StoryboardCommand<float>>();
+
+        public IReadOnlyList<StoryboardCommand<float>> Scale => scale;
+
+        private readonly SortedList<StoryboardCommand<Vector2>> vectorScale = new SortedList<StoryboardCommand<Vector2>>();
+
+        public IReadOnlyList<StoryboardCommand<Vector2>> VectorScale => vectorScale;
+
+        private readonly SortedList<StoryboardCommand<float>> rotation = new SortedList<StoryboardCommand<float>>();
+
+        public IReadOnlyList<StoryboardCommand<float>> Rotation => rotation;
+
+        private readonly SortedList<StoryboardCommand<Color4>> colour = new SortedList<StoryboardCommand<Color4>>();
+
+        public IReadOnlyList<StoryboardCommand<Color4>> Colour => colour;
+
+        private readonly SortedList<StoryboardCommand<float>> alpha = new SortedList<StoryboardCommand<float>>();
+
+        public IReadOnlyList<StoryboardCommand<float>> Alpha => alpha;
+
+        private readonly SortedList<StoryboardCommand<BlendingParameters>> blendingParameters = new SortedList<StoryboardCommand<BlendingParameters>>();
+
+        public IReadOnlyList<StoryboardCommand<BlendingParameters>> BlendingParameters => blendingParameters;
+
+        private readonly SortedList<StoryboardCommand<bool>> flipH = new SortedList<StoryboardCommand<bool>>();
+
+        public IReadOnlyList<StoryboardCommand<bool>> FlipH => flipH;
+
+        private readonly SortedList<StoryboardCommand<bool>> flipV = new SortedList<StoryboardCommand<bool>>();
+
+        public IReadOnlyList<StoryboardCommand<bool>> FlipV => flipV;
+
+        /// <summary>
+        /// Returns the earliest start time of the commands added to this group.
+        /// </summary>
+        [JsonIgnore]
+        public double StartTime { get; private set; } = double.MaxValue;
+
+        /// <summary>
+        /// Returns the latest end time of the commands added to this group.
+        /// </summary>
+        [JsonIgnore]
+        public double EndTime { get; private set; } = double.MinValue;
+
+        [JsonIgnore]
+        public double Duration => EndTime - StartTime;
```

Comment:
I'm not sure what to think of the fact that by default a `StoryboardCommandGroup` appears to have a `Duration` of negative infinity.

Reply:
yeah this has been a thing in master as well, I can change to null but then I have to handle null `StartTime`/`EndTime` all over the storyboard sprite class sooo.